web/packages/teleterm/src/services/pty/types.ts

```diff
@@ -78,6 +78,11 @@ export type GatewayCliClientCommand = PtyCommandBase & {
   env: Record<string, string>;
 };
 
+export type GatewayKubeCommand = PtyCommandBase & {
```

Comment:
`GatewayKubeCommand` and `ShellCommand` are in practice almost identical, they both open a new shell. The only difference between them is that `GatewayKubeCommand` sets `KUBECONFIG`.

There's a couple of places which should run the same logic when running a terminal, no matter if it's `GatewayKubeCommand` or `ShellCommand`, which currently match only on `cmd.kind === 'pty.shell'`.

Instead of adding checks for `cmd.kind === 'pty.gateway-kube'`, I think a better option might be to extend `ShellCommand` with `env?: Record<string, string>`.

Reply:
I recall the new type was avoiding `refreshTitile` from useDocumentTerminal.ts. what do you think

Reply:
I don't think we need to avoid refreshing the title for the kube tabs, but they already don't support showing proper cwd today and I don't want to pile too many refactors on you.

Still, I think implementing `GatewayKubeCommand` as `ShellCommand` instead is long-term a good idea. We could keep `refreshTitle` working only for regular terminal tabs by changing the conditional from `cmd.kind !== 'pty.shell'` to `doc.kind !== 'doc.terminal_shell'` and adding a TODO comment:

```typescript
// TODO(ravicious): Enable updating cwd in doc.gateway_kube titles by moving title-updating logic to DocumentsService.
```